### DISCUSSION TITLE
How to log from a benchmarkdotnet perf text

### DISCUSSION THREAD

**Discussion by originalfoo** [CitiesSkylinesMods/TMPE#1553] [Q&A] [answered]
I'm getting this error:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Benchmarks.SparseTaskListPerfTests.SparseTaskList_SimulatedCamMoves() in C:\Users\guyfr\Documents\GitHub\TMPE\TLM\Benchmarks\SparseTaskListPerfTests.cs:line 37
// ....
```

So naturally went to add some logging in the  benchmark method... but apparently that's not a thing. I just want to dump out some logging to identify which specific bit of line 37 is `null` to track down the cause of the problem.

**Comment by originalfoo** [comment, marked as answer]
Is there a diference between the net runtime of the benchmarks project and the rest of the mod?

I tried running the code from the benchmark (but with benchmark attribs removed, ie. just as a normal program) in dotnetfiddle and it works fine there.

**Reply by krzychu124** [reply]
Could you show the code? I found that benchmarks project results are only useful when you want to test if code generates garbage. For performance testing result is different - in game things will perform slightly worse and tests where you don't see any difference may actually give different results if you apply `AggressiveInline` attribute manually. Attribute does not seem to do anything while running project tests (results look like it is applied since when you apply `NoInlining` they run worse) - that's what for benchmark performs warmup before tests - to ensure that method will be compiled in a most optimal way (somewhat forcing inline if possible). 
There is another hidden attribute but I didn't check the mono source to see if it is also implemented but not exported in enum in the mscorlib dll or not
